tests/unit/AutoloadingTest.php

```diff
@@ -0,0 +1,108 @@
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\Payment_Gateway\Handlers;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\Payment_Gateway\PaymentFormContextChecker;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\REST_API;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\Settings_API;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Admin_Notice_Handler;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_Admin_Payment_Token_Editor;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_API_Authorization_Response;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_Apple_Pay_AJAX;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_Exception;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_Integration_Subscriptions;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_Payment_Tokens_Handler;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Plugin;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WP_Admin_Message_Handler;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WP_Job_Batch_Handler;
+use SkyVerge\WooCommerce\PluginFramework\v5_13_1\Tests\TestCase;
+
+/**
+ * @covers composer.json/autoload
+ */
+class AutoloadingTest extends TestCase
+{
+	public function testCanAutoload() : void
+	{
+		$list = [
+			Handlers\Capture::class,
+			Handlers\Abstract_Payment_Handler::class,
+			Handlers\Abstract_Hosted_Payment_Handler::class,
+			External_Checkout::class,
+			Admin::class,
+			Frontend::class,
+			Orders::class,
+			Google_Pay_Checkout\Google_Pay::class,
+			Google_Pay_Checkout\Admin::class,
+			Google_Pay_Checkout\AJAX::class,
+			Google_Pay_Checkout\Frontend::class,
+			Payment_Gateway_Blocks\Gateway_Blocks_Handler::class,
+			Payment_Gateway_Blocks\Gateway_Checkout_Block_Integration::class,
+			Settings_API\Abstract_Settings::class,
+			Settings_API\Setting::class,
+			Settings_API\Control::class,
+			Abstract_Cacheable_API_Base::class,
+			Notes_Helper::class,
+			Addresses\Address::class,
+			Addresses\Customer_Address::class,
+			SV_WC_Payment_Gateway_API_Authorization_Response::class,
+			SV_WC_Payment_Gateway_Admin_Payment_Token_Editor::class,
+			SV_WC_Payment_Gateway_Exception::class,
+			SV_WC_Payment_Gateway_Apple_Pay_AJAX::class,
+			SV_WC_Payment_Gateway_Integration_Subscriptions::class,
+			SV_WC_Payment_Gateway_Payment_Tokens_Handler::class,
+			\SkyVerge\WooCommerce\PluginFramework\v5_13_1\Payment_Gateway\REST_API::class,
+			REST_API::class,
+			REST_API\Controllers\Settings::class,
+			SV_WP_Job_Batch_Handler::class,
+			SV_WC_Admin_Notice_Handler::class,
+			SV_WP_Admin_Message_Handler::class,
+		];
+
+		Mockery::mock('\WP_REST_Controller');
+		Mockery::mock('\Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType');
+
+		foreach ($list as $className) {
+			$this->assertInstanceOf($className, Mockery::mock($className)->makePartial());
```

Comment:
Just curious: is this actually testing autoloading if we end up mocking the class? I'm not sure a class has to actually exist in order for this to work. For example, I adjusted the test like this and it still passed:

```diff
public function testCanAutoload() : void
{
	$list = [
		Handlers\Capture::class,
		Handlers\Abstract_Payment_Handler::class,
		Handlers\Abstract_Hosted_Payment_Handler::class,
		External_Checkout::class,
		Admin::class,
		Frontend::class,
		Orders::class,
		Google_Pay_Checkout\Google_Pay::class,
		Google_Pay_Checkout\Admin::class,
		Google_Pay_Checkout\AJAX::class,
		Google_Pay_Checkout\Frontend::class,
		Payment_Gateway_Blocks\Gateway_Blocks_Handler::class,
		Payment_Gateway_Blocks\Gateway_Checkout_Block_Integration::class,
		Settings_API\Abstract_Settings::class,
		Settings_API\Setting::class,
		Settings_API\Control::class,
		Abstract_Cacheable_API_Base::class,
		Notes_Helper::class,
		Addresses\Address::class,
		Addresses\Customer_Address::class,
		SV_WC_Payment_Gateway_API_Authorization_Response::class,
		SV_WC_Payment_Gateway_Admin_Payment_Token_Editor::class,
		SV_WC_Payment_Gateway_Exception::class,
		SV_WC_Payment_Gateway_Apple_Pay_AJAX::class,
		SV_WC_Payment_Gateway_Integration_Subscriptions::class,
		SV_WC_Payment_Gateway_Payment_Tokens_Handler::class,
		\SkyVerge\WooCommerce\PluginFramework\v5_13_1\Payment_Gateway\REST_API::class,
		REST_API::class,
		REST_API\Controllers\Settings::class,
		SV_WP_Job_Batch_Handler::class,
		SV_WC_Admin_Notice_Handler::class,
		SV_WP_Admin_Message_Handler::class,
+		'ClassThatDoesNotExist',
	];

	Mockery::mock('\WP_REST_Controller');
	Mockery::mock('\Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType');

	foreach ($list as $className) {
		$this->assertInstanceOf($className, Mockery::mock($className)->makePartial());
	}
}
```

What do you think about this instead?

```php
	/**
	 * @dataProvider providerCanAutoload
	 */
	public function testCanAutoload(string $className) : void
	{
		Mockery::mock('\WP_REST_Controller');
		Mockery::mock('\Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType');

		$this->assertTrue(class_exists($className));
	}

	public function providerCanAutoload(): \Generator
	{
		$list = [
			Handlers\Capture::class,
			Handlers\Abstract_Payment_Handler::class,
			Handlers\Abstract_Hosted_Payment_Handler::class,
			External_Checkout::class,
			Admin::class,
			Frontend::class,
			Orders::class,
			Google_Pay_Checkout\Google_Pay::class,
			Google_Pay_Checkout\Admin::class,
			Google_Pay_Checkout\AJAX::class,
			Google_Pay_Checkout\Frontend::class,
			Payment_Gateway_Blocks\Gateway_Blocks_Handler::class,
			Payment_Gateway_Blocks\Gateway_Checkout_Block_Integration::class,
			Settings_API\Abstract_Settings::class,
			Settings_API\Setting::class,
			Settings_API\Control::class,
			Abstract_Cacheable_API_Base::class,
			Notes_Helper::class,
			Addresses\Address::class,
			Addresses\Customer_Address::class,
			SV_WC_Payment_Gateway_API_Authorization_Response::class,
			SV_WC_Payment_Gateway_Admin_Payment_Token_Editor::class,
			SV_WC_Payment_Gateway_Exception::class,
			SV_WC_Payment_Gateway_Apple_Pay_AJAX::class,
			SV_WC_Payment_Gateway_Integration_Subscriptions::class,
			SV_WC_Payment_Gateway_Payment_Tokens_Handler::class,
			\SkyVerge\WooCommerce\PluginFramework\v5_13_1\Payment_Gateway\REST_API::class,
			REST_API::class,
			REST_API\Controllers\Settings::class,
			SV_WP_Job_Batch_Handler::class,
			SV_WC_Admin_Notice_Handler::class,
			SV_WP_Admin_Message_Handler::class,
		];

		foreach($list as $className) {
			yield $className => [$className];
		}
	}
```

1. It uses `class_exists()` -- I believe that triggers autoloading.
2. It uses a data provider, which just gives us more insight into failures (if one fails, it tells us which one).

This change is actually resulting in an error for me:

> There was 1 failure:
> 
> 1) SkyVerge\WooCommerce\PluginFramework\v5_13_1\Tests\Unit\AutoloadingTest::testCanAutoload with data set "SkyVerge\WooCommerce\PluginFramework\v5_13_1\SV_WC_Payment_Gateway_API_Authorization_Response" ('SkyVerge\WooCommerce\PluginFr...sponse')
> Failed asserting that false is true.

Reply:
That is weird, on my local setup when I tried the same case and added non-existing class, it failed! Based on my understanding `Mockery::mock($className)->makePartial()` only works with defined classes. But I will update it similar to your example for consistency as it makes sense to check for the class existence that triggers the autoload as you mentioned

Reply:
Done => [67e69bc](https://github.com/godaddy-wordpress/wc-plugin-framework/pull/704/commits/67e69bc4e6ead2c22e8d424fb7291f9c6fc56715)

Reply:
Looks great, thank you!